gateway/ln-gateway/tests/integration_tests.rs

```diff
@@ -1167,6 +1167,82 @@ async fn test_gateway_shows_info_about_all_connected_federations() -> anyhow::Re
     .await
 }
 
+#[tokio::test(flavor = "multi_thread")]
+async fn test_gateway_can_leave_connected_federations() -> anyhow::Result<()> {
```

Comment:
Should we also add a devimint test that reboots the gateway in between to verify that the short channel ids are correct after the restart (or maybe the existing devimint test can be modified to cover this)?

Reply:
Yea the bug we observed *only* happened after restarting the gateway.

Reply:
The duplicated id was certainly after restarting. But we also had issues (same? others?) before restarting.

Reply:
Adding these tests highlighted several bugs / slight logical with scid assignments. If we backported this to 0.2.1 and updated a live gateway, future scid assignments when connecting / disconnecting feds would be predictable as defined by the new test spec.

> Tricky thing is that we may need at least 2 federations for this devimint test?

Yes. We have a [gateway reboot test](https://github.com/fedimint/fedimint/blob/412aaf965805263f397364242618df5a27cc7b58/devimint/src/tests.rs#L1342), but we can't improve it's coverage to scid assignments because devimint is limited. Relevant to #3985

> We can't really tests a gatewayd restart in this test because it uses memdb?

Well this also is a limitation for testing persisted states